core/src/test/java/org/apache/iceberg/MetadataTestUtils.java

```diff
@@ -0,0 +1,428 @@
+      return this;
+    }
+
+    public TableMetadataBuilder setPartitionStatisticsFiles(
+        List<PartitionStatisticsFile> partitionStatisticsFiles) {
+      this.partitionStatisticsFiles = partitionStatisticsFiles;
+      return this;
+    }
+
+    public TableMetadata build() {
+      return new TableMetadata(
+          metadataLocation,
+          formatVersion,
+          uuid,
+          location,
+          lastSequenceNumber,
+          lastUpdatedMillis,
+          lastColumnId,
+          currentSchemaId,
+          ImmutableList.copyOf(schemas),
+          defaultSpecId,
+          ImmutableList.copyOf(specs),
+          lastAssignedPartitionId,
+          defaultSortOrderId,
+          ImmutableList.copyOf(sortOrders),
+          ImmutableMap.copyOf(properties),
+          currentSnapshotId,
+          ImmutableList.copyOf(snapshots),
+          snapshotsSupplier,
+          ImmutableList.copyOf(snapshotLog),
+          ImmutableList.copyOf(previousFiles),
+          ImmutableMap.copyOf(refs),
+          ImmutableList.copyOf(statisticsFiles),
+          ImmutableList.copyOf(partitionStatisticsFiles),
+          ImmutableList.copyOf(changes));
+    }
+  }
+
+  public static BaseSnapshotBuilder buildTestSnapshot() {
+    return new BaseSnapshotBuilder();
+  }
+
+  public static BaseSnapshotBuilder buildTestSnapshotWithExampleValues() {
+    return new BaseSnapshotBuilder()
+        .setSequenceNumber(EXAMPLE_SEQUENCE_NUMBER)
+        .setSnapshotId(EXAMPLE_SNAPSHOT_ID)
+        .setParentId(EXAMPLE_PARENT_ID)
+        .setOperation(DataOperations.REPLACE)
+        .setSummary(ImmutableMap.of("files-added", "4", "files-deleted", "100"))
+        .setSchemaId(EXAMPLE_SCHEMA_ID)
+        .setTimestampMillis(System.currentTimeMillis());
+  }
+
+  public static class BaseSnapshotBuilder {
+    private long snapshotId;
+    private Long parentId;
+    private long sequenceNumber;
+    private long timestampMillis;
+    private String manifestListLocation;
+    private String operation;
+    private Map<String, String> summary;
+    private Integer schemaId;
+    private String[] v1ManifestLocations;
+
+    private BaseSnapshotBuilder() {
+      this.snapshotId = -1L;
+      this.sequenceNumber = -1L;
+      this.timestampMillis = System.currentTimeMillis();
+      this.summary = ImmutableMap.of();
+    }
+
+    public BaseSnapshotBuilder setSnapshotId(long snapshotId) {
+      this.snapshotId = snapshotId;
+      return this;
+    }
+
+    public BaseSnapshotBuilder setParentId(Long parentId) {
+      this.parentId = parentId;
+      return this;
+    }
+
+    public BaseSnapshotBuilder setSequenceNumber(long sequenceNumber) {
+      this.sequenceNumber = sequenceNumber;
+      return this;
+    }
+
+    public BaseSnapshotBuilder setTimestampMillis(long timestampMillis) {
+      this.timestampMillis = timestampMillis;
+      return this;
+    }
+
+    public BaseSnapshotBuilder setManifestListLocation(String manifestListLocation) {
```

Comment:
What about making this similar to `setV1ManifestLocations` instead and combining it with `buildWithExampleManifestList`? I think that would make more sense.